pandera/engines/polars_engine.py

```diff
@@ -0,0 +1,506 @@
+            return (
+                (self.type == pandera_dtype.type)
+                & (self.scale == pandera_dtype.scale)
+                & (self.precision == pandera_dtype.precision)
+            )
+
+        except TypeError:  # pragma: no cover
+            return super().check(pandera_dtype)
+
+    def __str__(self) -> str:
+        return f"Decimal(precision={self.precision}, scale={self.scale})"
+
+
+###############################################################################
+# Temporal types
+###############################################################################
+
+
+@Engine.register_dtype(
+    equivalents=[
+        "date",
+        datetime.date,
+        pl.Date,
+        dtypes.Date,
+        dtypes.Date(),
+    ]
+)
+@immutable
+class Date(DataType, dtypes.Date):
+    """Polars date data type."""
+
+    type = pl.Date
+
+
+@Engine.register_dtype(
+    equivalents=[
+        "datetime",
+        datetime.datetime,
+        pl.Datetime,
+        dtypes.DateTime,
+        dtypes.DateTime(),
+    ]
+)
+@immutable(init=True)
+class DateTime(DataType, dtypes.DateTime):
+    """Polars datetime data type."""
+
+    type = pl.Datetime
+
+    def __init__(  # pylint:disable=super-init-not-called
+        self,
+        time_zone: Optional[str] = None,
+        time_unit: Optional[str] = None,
+    ) -> None:
+        object.__setattr__(self, "type", pl.Datetime(time_zone, time_unit))
+
+    @classmethod
+    def from_parametrized_dtype(cls, polars_dtype: pl.Datetime):
+        """Convert a :class:`polars.Decimal` to
+        a Pandera :class:`pandera.engines.polars_engine.Decimal`."""
+        return cls(
+            time_zone=polars_dtype.time_zone, time_unit=polars_dtype.time_unit
+        )
+
+
+@Engine.register_dtype(
+    equivalents=[
+        "time",
+        datetime.time,
+        pl.Time,
+    ]
+)
+@immutable
+class Time(DataType):
+    """Polars time data type."""
+
+    type = pl.Time
+
+
+@Engine.register_dtype(
+    equivalents=[
+        "timedelta",
+        datetime.timedelta,
+        pl.Duration,
+        dtypes.Timedelta,
+        dtypes.Timedelta(),
+    ]
+)
+@immutable(init=True)
+class Timedelta(DataType, dtypes.Timedelta):
+    """Polars timedelta data type."""
+
+    type = pl.Duration
+
+    def __init__(  # pylint:disable=super-init-not-called
+        self,
+        time_unit: Literal["ns", "us", "ms"] = "us",
+    ) -> None:
+        object.__setattr__(self, "type", pl.Duration(time_unit))
+
+    @classmethod
+    def from_parametrized_dtype(cls, polars_dtype: pl.Duration):
+        """Convert a :class:`polars.Duration` to
+        a Pandera :class:`pandera.engines.polars_engine.Duration`."""
+        if polars_dtype.time_unit is None:
+            return cls()
+        return cls(time_unit=polars_dtype.time_unit)
+
+
+###############################################################################
+# Nested types
+###############################################################################
```

Comment:
is this a placeholder comment for nested types? https://docs.pola.rs/py-polars/html/reference/datatypes.html#nested

Would this be implemented in a follow-up PR?

Reply:
Yes